articles/building-an-appointment-scheduling-telegram-bot/index.md

```diff
@@ -0,0 +1,514 @@
+
+#### Creating a Fauna Index
+
+You will need to create a Fauna index that will allow you to scroll through the data created in your database. To do this, go to the `DB Overview` tab on the left side of your screen, then click on the `New Index` button.
+
+For this article, you will be creating three indexes; `users_index`, `appointment_index`, and `appointment_today_index`. The `users_index` index will enable you to scroll through data in the `Users` collection using the `id` field as a parameter to match. The `appointment_index` index will enable you to scroll through data in the `Appointments` collection using the `user_id` field as a parameter to match. The `appointment_today_index` index will allow you to scroll through data in the `Appointments` collection. It will use the `user_id` and `date_due` fields as parameters to match.
+
+![create_index](/engineering-education/building-an-appointment-scheduling-telegram-bot/create_index.png)
+
+After clicking on the `New Index` button, the system will display the screen above to you. You need to select the collection you want to connect to this particular index. After choosing the collection, enter a name for your index, terms for your index, and values. Ticking the `Unique` checkbox ensures the data entered for the term is unique. The terms field specifies what data you want the index to be able to browse.
+
+For the `users_index`, we will use the `id` field as the terms. Also make sure you tick the unique checkbox so the data entered is unique. 
+For the `apppointment_index`, we will use the `user_id` field as the terms. For the `appointment_today_index` we will use the `user_id` and `date_due` fields as the terms. After filling in the required fields, click on save and continue.
+
+![indexes](/engineering-education/building-an-appointment-scheduling-telegram-bot/indexes.png)
+
+### Integrating Fauna with Python
+
+#### Creating a Fauna API Key
+
+For your python app to interact with fauna, you need to create an API key for the database. To do this, go to the security tab on the left side of your screen.
+
+![new_key](/engineering-education/building-an-appointment-scheduling-telegram-bot/new_key.png)
+
+Click on `New Key` to generate a secret key. To create the key, you will need to provide a name for your key and save it. Your secret key will now be displayed to you, as seen in the image below. Copy and paste it somewhere you can easily retrieve it.
+
+![key](/engineering-education/building-an-appointment-scheduling-telegram-bot/key.png)
+
+#### Prerequisites
+
+From this point onwards, to follow this step by step guide on building an appointment scheduling telegram bot with fauna, you need to have the following installed:
+
+
+* [Python 3.7 or >3.7](http://python.org/)
+* [Faunadb](https://pypi.org/project/faunadb/)
+* [telegram](https://pypi.org/project/telegram/)
+* [python_telegram_bot](https://pypi.org/project/python-telegram-bot/)
+
```

Comment:
Move this section to the beginning of the article (after the introduction part).
Put a dot at the end of every line.
Change the first line to:
To follow along with this tutorial, you need to have: